service/src/event/service.rs

```diff
@@ -237,6 +241,168 @@ impl CeramicEventService {
+                            },
+                            previous: data.prev().iter().copied().collect(),
+                            data: data.content().to_vec(),
+                            index: 0, 
+                        }))
+                    }
+                    ceramic_event::unvalidated::Payload::Init(init) => {
+                        Ok(ConclusionEvent::Data(ConclusionData {
+                            event_cid: raw_event.id,
+                            init: ConclusionInit {
+                                stream_cid: raw_event.id, // For init events, the event CID is the stream CID
+                                stream_type: init.header().stream_type() as u8,
+                                controller: init
+                                    .header()
+                                    .controllers()
+                                    .first()
+                                    .unwrap()
+                                    .to_string(),
+                                dimensions: vec![], //TODO_Discuss: You may need to populate this if needed
+                            },
+                            previous: vec![],
+                            data: init.content().to_vec(),
+                            index: 0,
+                        }))
+                    }
+                }
+            }
+
+            ceramic_event::unvalidated::Event::Unsigned(unsigned_event) => {
+                Ok(ConclusionEvent::Data(ConclusionData {
+                    event_cid: raw_event.id,
+                    init: ConclusionInit {
+                        stream_cid: raw_event.id, // For unsigned events, the event CID is the stream CID
+                        stream_type: unsigned_event.header().stream_type() as u8,
+                        controller: unsigned_event
+                            .header()
+                            .controllers()
+                            .first()
+                            .unwrap()
+                            .to_string(),
+                        dimensions: vec![], // You may need to populate this if needed
+                    },
+                    previous: vec![],
+                    data: unsigned_event.content().to_vec(),
+                    index: 0,
+                }))
+            }
+        }
+    }
+
+    // Helper method to get an event by its CID
+    fn get_event_by_cid(&self, cid: &Cid) -> Result<ceramic_event::unvalidated::Event<Ipld>> {
+        // Implement this method to fetch the event from your storage
+        unimplemented!("Implement get_event_by_cid method")
+    }
+
+    async fn conclusion_events_since(
+        &self,
+        highwater_mark: i64,
+        limit: i64,
+        include_data: ceramic_api::IncludeEventData,
+    ) -> anyhow::Result<Vec<ConclusionEvent>> {
+        let raw_events = ceramic_api::EventStore::events_since_highwater_mark(
```

Comment:
We do not want to rely on the API methods here. We should likely move the body of `ceramic_api::EventStore::events_since_highwater_mark` to a method on the struct directly. Both this method and the EventStore implementation can call it. This way we do not overload the ceramic_api types.